components/plans/create/AttributePicker.tsx

```diff
@@ -0,0 +1,231 @@
+import { PlusIcon, XMarkIcon } from "@heroicons/react/20/solid";
+import clsx from "clsx";
+import { useState } from "react";
+import { useMap } from "react-map-gl";
+import Button from "../../Button";
+import CardSlider from "../../CardSlider";
+
+interface AttributesPickerProps {
+  attributes: string[] | undefined;
+  selectedAttributes: string[][];
+  setSelectedAttributes: (stops: string[][]) => void;
+  attributesIndex: number;
+  setAttributesIndex: (index: number) => void;
+  setSelectedTabIndex: (index: number) => void;
+  suggestions: Venue[] | undefined;
+}
+
+export default function AttributesPicker({
+  attributes,
+  selectedAttributes,
+  setSelectedAttributes,
+  attributesIndex,
+  setAttributesIndex,
+  setSelectedTabIndex,
+  suggestions,
+}: AttributesPickerProps) {
+  const { map } = useMap();
+
+  const addStop = (stopIndex: number) => {
```

Comment:
This sort of thinking is something that is good too abstract into a hook. We can then test the functions a little more independently in the future if we so wish